### PR TITLE
add a version for the pattern fingerprint

### PR DESCRIPTION
**Pull request by greglandrum**
this should have been in there long ago

### CODE REVIEW

File: Code/GraphMol/Fingerprints/Fingerprints.h [hunk 1]
```diff
@@ -146,6 +146,7 @@ RDKIT_FINGERPRINTS_EXPORT ExplicitBitVect *PatternFingerprintMol(
     std::vector<unsigned int> *atomCounts = nullptr,
     ExplicitBitVect *setOnlyBits = nullptr,
     bool tautomericFingerprint = false);
+const std::string PatternFingerprintMolVersion = "1.0.0";
```
Comment (jones-gareth) [inline Code/GraphMol/Fingerprints/Fingerprints.h:149]: A string containing major, minor and patch levels seems overkill for the pattern fingerprinting version.   I think an integer number would be sufficient and easier to parse.

If you want to use the current version string maybe split out the version numbers, so we can guarantee the format in the future:

```c++
const int major = 1;
const int minor = 0;
const int patch = 0
const std::string PatternFingerprintMolVersion = to_string(major)+'.'+ .... ;
```

Comment (greglandrum) [reply Code/GraphMol/Fingerprints/Fingerprints.h:149]: > A string containing major, minor and patch levels seems overkill for the pattern fingerprinting version. I think an integer number would be sufficient and easier to parse.

Agreed, but this is consistent with the other version numbers that we have for toolkits and descriptors. It's worth thinking about how to change that in order to make it easier to parse/handle, but that's a longer conversation. Maybe that kind of change would be something for the next release cycle

Comment (jones-gareth) [reply Code/GraphMol/Fingerprints/Fingerprints.h:149]: Makes sense.

**Comment by bp-kelley** [comment]
@greglandrum A comment I made didn't stick apparently.  From the perspective of ChemFP this would be pattern/2 as there was a bug-fix change a while back that changed the fps.

It might be nice to sync these?

**Comment by greglandrum** [comment]
@bp-kelley There have been multiple changes over the years, so we just have to pick something to call 1.0. 
I don’t really care that much, if it’s better to switch to 2.0, I’m fine with that

**Comment by bp-kelley** [comment]
@greglandrum I'm not sure that Andrew would trust the version # anyway, so 1 it is!

**Comment by jones-gareth** [comment]
@greglandrum Is it possible to do something similar for Morgan fingerprints and smiles canonicalization? I know the smiles canonicalization may not be practical, but versioning Morgan fingerprints might be.

**Comment by greglandrum** [comment]
@jones-gareth : it's already there for the Morgan FP: https://github.com/rdkit/rdkit/blob/master/Code/GraphMol/Fingerprints/MorganFingerprints.h#L58

SMILES canonicalization is more challenging and I think can really only be done via the regression-testing based approach we discussed elsewhere

**Comment by jones-gareth** [comment]
@greglandrum Thanks- sorry I thought I had looked for the version number for Morgan fingerprints, but I guess I didn't look too hard.  I saw your email on regression testing- do you think it would be a good idea to expose the changes identified in the regression tests through the API?

**Pull request merged**